src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/SingleShot.cs

```diff
@@ -179,6 +179,61 @@ public void SetTimeout(int ms)
             NNStreamer.CheckException(ret, "fail to set the timeout!");
         }
 
+        /// <summary> Sets the property value for the given model. 
+        /// <para> A model/framework may support changing the model information, such as tensor dimension and data layout, after opening the model. </para>
+        /// <para> If tries to change unavailable property or the model do not allow changing the model information, this will raise an exception. </para>
+        /// <para> For the details about the properties, see 'tensor_filter' plugin definition in NNStreamer. </para>
+        /// <see cref="https://github.com/nnsuite/nnstreamer"/>.
+        /// </summary>
+        /// <param name="name">The property name</param>
+        /// <param name="value">The property value</param>
+        /// <feature>http://tizen.org/feature/machine_learning.inference</feature>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="ArgumentException">Thrown when the method failed due to an invalid parameter.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public void SetValue(string name, string value)
+        {
+            NNStreamerError ret = NNStreamerError.None;
+
+            NNStreamer.CheckNNStreamerSupport();
+
+            /* Check the argument */
+            if (string.IsNullOrEmpty(name))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property name is invalid");
+
+            if (string.IsNullOrEmpty(value))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property value is invalid");
+
+            ret = Interop.SingleShot.SetValue(_handle, name, value);
+            NNStreamer.CheckException(ret, "Failed to to set the property value");
```

Comment:
How about update exception message?
`Not supported property name`

Reply:
I'd like to distinguish between invalid property key error and invalid value error.

Reply:
add detailed exception message for unavailable property and invalid value